homeassistant/components/smartenergy_awattar/coordinator.py

```diff
@@ -0,0 +1,63 @@
+"""The Awattar coordinator."""
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+import logging
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import API, DOMAIN, INIT_STATE
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+
+class AwattarCoordinator(DataUpdateCoordinator):
```

Comment:
```suggestion
class AwattarCoordinator(DataUpdateCoordinator[data_type]):
```

Change `data_type` to the new class, which holds the data